odoo/sql_db.py

```diff
@@ -279,6 +279,9 @@ def dictfetchmany(self, size):
         return [self.__build_dict(row) for row in self._obj.fetchmany(size)]
     def dictfetchall(self):
         return [self.__build_dict(row) for row in self._obj.fetchall()]
+    def batched_dictfetchall(self, size):
+        for _ in range(0, int(self._obj.rowcount/size) + 1):
+            yield self.dictfetchmany(size)
```

Comment:
Hi,

Not sure that it is necessary to add a method for this. We normally make a generic method only if there are multiple example where it can be useful.

Side note: also, if you have memory error, try to use `fetchall`/`fetchmany` instead of `dictfetchall` (AFAIK, it takes +-3 times less memory in my last test).